app/src/main/java/kth/jjve/xfran/HomeScreenActivity.java

```diff
@@ -20,6 +32,27 @@ protected void onCreate(Bundle savedInstanceState) {
         FrameLayout contentFrameLayout = findViewById(R.id.content_frame);
         getLayoutInflater().inflate(R.layout.act_home, contentFrameLayout);
 
+        /*----- HOOKS -----*/
+        WODTitle = findViewById(R.id.home_WODTitle);
+        WODDescription = findViewById(R.id.home_WODDescription);
+        WODExercises = findViewById(R.id.home_WODExercises);
+        Username = findViewById(R.id.home_username);
+        PlannedWorkouts = findViewById(R.id.home_rv);
+
+        /*-----  VM  -----*/
+        homeVM = ViewModelProviders.of(this).get(HomeVM.class);
+        homeVM.init(getApplicationContext());
+        homeVM.getUserName().observe(this, s -> Username.setText(s));
```

Comment:
maybe there's a bug bcos the user name does not appear on home screen when i run it

Reply:
Whatever I try, I am not able to replicate the bug you are talking about. So either its weird on your phone, or you are not seeing it correctly. I will message you